lib/Dialect/Ibis/IbisOps.cpp

```diff
@@ -262,6 +237,26 @@ LogicalResult GetPortOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// ThisOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult ThisOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  // A thisOp should always refer to the parent operation, which in turn should
+  // be an Ibis ScopeOpInterface.
+  auto parentScope =
+      dyn_cast_or_null<ScopeOpInterface>(getOperation()->getParentOp());
+  if (!parentScope)
+    return emitOpError() << "thisOp must be nested in a scope op";
```

Comment:
Can you express this in ODS with `HasParent<ScopeOpInterface>`?

Reply:
Nope, generated code expects a `static getOperationName` to be defined for the type.